core/flyout_base.js

```diff
@@ -615,9 +616,28 @@ Blockly.Flyout.prototype.createBlock = function(originalBlock) {
     Blockly.Events.enable();
   }
 
+  var variablesAfterCreation = this.targetWorkspace_.getAllVariables();
```

Comment:
Make this into a helper function: `Blockly.Flyout.prototype.getAddedVariables(variablesBeforeCreation)` that returns the list of new variables.

Reply:
done.